src/app_service/service/contacts/service.nim

```diff
@@ -678,16 +680,23 @@ QtObject:
 
   proc cancelVerificationRequest*(self: Service, publicKey: string) =
     try:
-      let response = status_contacts.cancelVerificationRequest(publicKey)
-      if(not response.error.isNil):
+      var response = status_contacts.getVerificationRequestSentTo(publicKey)
+      if not response.error.isNil:
+        let msg = response.error.message
+        raise newException(RpcException, msg)
+
+      let request = response.result.toVerificationRequest()
+
+      response = status_contacts.cancelVerificationRequest(request.id)
+      if not response.error.isNil:
         let msg = response.error.message
         error "error sending contact verification request", msg
         return
 
       var contact = self.getContactById(publicKey)
       contact.verificationStatus = VerificationStatus.Unverified
       self.saveContact(contact)
```

Comment:
@jrainville do we need to remove canceled request from the map? What should it present active verification request or latest, regardless  is it active or not